command/cp.go

```diff
@@ -367,7 +380,14 @@ func (c Copy) doDownload(ctx context.Context, srcurl *url.URL, dsturl *url.URL)
 		return err
 	}
 
-	f, err := os.Create(dsturl.Absolute())
+	r, err := dstClient.Make(ctx, storage.MakeOpts{
```

Comment:
MakeOpts is unnecessary. `path` is the mandatory argument, and only `directory` is optional. This is better:

`func (fs *Filesystem) Make(ctx context.Context, path string, isDirectory bool)`

Another thing to not is that, I still don't like this method because it does 3 whole different things, and for only 1 scenario caller has to call Close method. I'd prefer 3 methods, each one doing one thing (and seem obvious from the signature). Let's merge this as it is and talk about it for the next one.

Reply:
hmm, ok, thanks.